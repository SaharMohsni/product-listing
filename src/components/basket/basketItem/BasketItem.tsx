import React from 'react';
import { useDispatch } from 'react-redux';
import { IInBasketProduct } from '../../../features/types/products.types';
import './basket-item.css';

interface IOwnProps {
	item: IInBasketProduct;
}

const BasketItem: React.FC<IOwnProps> = ({ item }) => {
	return (
		<div className="basket-item global-flex-h-between-v-center">
			<div className="basket-item__details">
				<div className="basket-item__details__name product-name">{item.productData.name}</div>
				<div className="basket-item__details__price-details product-price">
					<span className="basket-item__details__price-details__unit product-price__unit">$</span>
					<span className="basket-item__details__price-details__price product-price__price">
						{item.productData.price}
					</span>
				</div>
			</div>
			<div className="basket-item__quantity-details global-flex-h-between-v-center">
				<button className="basket-item__quantity-details__increment-btn global-flex-h-center-v-center">
					+
				</button>
				<div className="basket-item__quantity-details__quantity global-flex-h-center-v-center">
					{item.quantity}
				</div>
				<button className="basket-item__quantity-details__decrement-btn global-flex-h-center-v-center">
					-
				</button>
			</div>
		</div>
	);
};

export default BasketItem;
