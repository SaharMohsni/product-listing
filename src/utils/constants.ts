export const ASC_PRICE = 0;
export const DESC_PRICE = 1;
export const NEW_ADDED = 2;
export const OLD_ADDED = 3;

export const LIMIT_PRODUCTS_BY_Page = 16;

export const sortDataMessages = {
	[ASC_PRICE]: 'Price low to high',
	[DESC_PRICE]: 'Price high to low',
	[NEW_ADDED]: 'New to old',
	[OLD_ADDED]: 'Old to new'
};

export const MIN_PRODUCT_QUANTITY_IN_BASKET = 1;
export const FAKE_DATA = Array.from({ length: 16 }).map(() => {}).map((el, i) => ({
	tags: [ '' ],
	price: i,
	name: 'string',
	description: 'string',
	slug: 'string',
	added: i,
	manufacturer: i,
	itemType: 'string'
}));
export const DELAY_TIME = 3000;
