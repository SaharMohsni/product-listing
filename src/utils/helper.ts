import { ISearchProductsPayload } from '../features/types/products.types';
import { isEmpty } from 'lodash';
export const getURLCleanPath = (pathname: string) => pathname.split('/')[1];

export const serialize = function(obj: any) {
	var str = [];
	for (var p in obj)
		if (obj.hasOwnProperty(p)) {
			str.push(encodeURIComponent(p) + '=' + encodeURIComponent(obj[p]));
		}
	return str.join('&');
};

export const handleNavigationQuery = (params: ISearchProductsPayload) => {
	let { page, sortVariable, sortType } = params;
	let query = params;
	if (!isEmpty(params) && (isEmpty(sortVariable) || isEmpty(sortType))) {
		query = {
			page: page
		};
	} else {
		query = {
			...params
		};
	}

	return `${serialize(query)}`;
};
