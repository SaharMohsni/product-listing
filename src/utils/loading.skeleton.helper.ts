export const fullRowItemSkeleton = (loading:boolean, active = true) => ({
    active,
    title: { rows: 1, width: '100%' },
    paragraph: false,
    loading,
  })
  export const labelSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 1, width: '20%'  },
    loading,
  })
  export const tagSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 1 },
    loading,
  })
  export const imageSkeleton = (loading:boolean, active = true ) => ({
    loading,
  active,
  paragraph: false,
  title:false
  })

  export const descriptionSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 3, width: '100%' },
    loading,
  })
  export const shapeSquareBoxSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: false,
    loading,
  })
