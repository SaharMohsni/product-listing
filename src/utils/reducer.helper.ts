import { ISearchProductsPayload } from '../features/types/products.types';

export const handleSearchParams = (localSearchParams: ISearchProductsPayload, params: ISearchProductsPayload) => {
	let { page, limit, sortVariable, sortType } = localSearchParams;
	return {
		...localSearchParams,
		page: params.page ? params.page : page,
		limit: params.limit ? params.limit : limit,
		sortVariable: params.sortVariable ? params.sortVariable : sortVariable,
		sortType: params.sortType ? params.sortType : sortType
	};
};
