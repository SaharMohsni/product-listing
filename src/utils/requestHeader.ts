 
export const requestHeader = (options: object| null ): object => ({
  headers: { 
    // 'Access-Control-Allow-Origin': '*', 
    ...options,
  },
})

 