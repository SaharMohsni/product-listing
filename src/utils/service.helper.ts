import { isEmpty } from 'lodash';
export const addSortToUrl = (variable: string | undefined, sortType: string | undefined) => {
	if (!isEmpty(variable) && !isEmpty(sortType)) {
		return `&_sort=${variable}&_order=${sortType}`;
	} else return '';
};

export const addFilterToUrl = (variable: any | undefined) => {
	let str = [];
	if (variable) {
		for (var el in variable.data) {
			str.push(variable.key + '=' + variable.data[el]);
		}
	}
	return str.join('&');
};

export const addFilterByArrayListToUrl = (variable: any | undefined) => {
	let str = [];
	if (variable) {
		for (var el in variable.data) {
			str.push(variable.key + '_like=' + variable.data[el]);
		}
	}
	return str.join('&');
};

export const addFilterByStringToUrl = (variable: any | undefined) => {
	if (!isEmpty(variable)) {
		return `itemType=${variable}`;
	} else return '';
};
