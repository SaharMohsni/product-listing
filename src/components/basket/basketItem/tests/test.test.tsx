import React from 'react';
import { render } from '../../../../utils/testing';
import BasketItem, { IOwnProps } from '../BasketItem';

describe('BasketItem', () => {
	const defaultProps: IOwnProps = {
		item: {
			productData: {
				tags: [ '' ],
				price: 0,
				name: '',
				description: '',
				slug: '',
				added: 0,
				manufacturer: 0,
				itemType: ''
			},
			quantity: 1
		}
	};
	const { container } = render(<BasketItem {...defaultProps} />);
	test('should render the basic template', async () => {
		expect(container).toMatchSnapshot();
	});
});
