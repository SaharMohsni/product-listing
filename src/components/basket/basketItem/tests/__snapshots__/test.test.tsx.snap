// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketItem should render the basic template 1`] = `
<div>
  <div
    class=" basket-item global-flex-h-between-v-center"
  >
    <div
      class="basket-item__details"
    >
      <div
        class="basket-item__details__name product-name"
      />
      <div
        class="basket-item__details__price-details product-price"
      >
        <span
          class="basket-item__details__price-details__unit product-price__unit"
        >
          $
        </span>
        <span
          class="basket-item__details__price-details__price product-price__price"
        >
          0
        </span>
      </div>
    </div>
    <div
      class="basket-item__quantity-details global-flex-h-between-v-center"
    >
      <button
        class="basket-item__quantity-details__increment-btn global-flex-h-center-v-center"
      >
        +
      </button>
      <div
        class="basket-item__quantity-details__quantity global-flex-h-center-v-center"
      >
        1
      </div>
      <button
        class="basket-item__quantity-details__decrement-btn global-flex-h-center-v-center"
      >
        -
      </button>
    </div>
  </div>
</div>
`;
